components/main_menu/main_menu.jsx

```diff
@@ -138,6 +140,18 @@ class MainMenu extends React.PureComponent {
             );
         });
 
+        const integrationItems = this.props.integrationMenuItems.map((item) => {
+            return (
+                <Menu.ItemAction
+                    id={item.id + '_pluginmenuitem'}
```

Comment:
What value is this `id` property. Dynamically generated or hard coded in the plugin's config for example?

Reply:
That id is the plugin id. I know that means that the same plugin cannot create two different actions for the same location without react complaining about repeated keys. I have mixed feelings about using unique ids for the actions, but we can discuss it.